remote-sources/kubernetes-fs/main.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/icza/dyno"
+	"gopkg.in/yaml.v2"
+)
+
+type ObjectMeta struct {
+	Name      string `yaml:"name"`
+	Namespace string `yaml:"namespace"`
+}
+
+type AgogosResource struct {
+	Kind       string             `yaml:"kind"`
+	APIVersion string             `yaml:"apiVersion"`
+	Metadata   ObjectMeta         `yaml:"metadata"`
+	Spec       AgogosResourceSpec `yaml:"spec"`
+}
+
+type AgogosResourceSpec interface{}
+
+func (r AgogosResource) FullName() string {
+	var namespace string = r.Metadata.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	return fmt.Sprintf("%s_%s", namespace, r.Metadata.Name)
+}
+
+func readResources(dirPath string) ([]AgogosResource, error) {
+	files, err := ioutil.ReadDir(dirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]AgogosResource, len(files))
+
+	for i, file := range files {
+		content, err := os.Open(path.Join(dirPath, file.Name()))
+		if err != nil {
+			return nil, err
+		}
+
+		defer content.Close()
+
+		decoder := yaml.NewDecoder(content)
+		var resource AgogosResource
+		decoder.Decode(&resource)
+		resources[i] = resource
+	}
+
+	return resources, nil
+}
+
+var unknownKind = "UNKNOWN_KIND"
+
+func toIndex(resource AgogosResource) string {
+	switch resource.Kind {
+	case "UpstreamClientCredentials":
+		return "upstreamclientcredentials"
+	case "Upstream":
+		return "upstreams"
+	case "Schema":
```

Comment:
Might be worth putting these mappings in a map and returning `unknownKind` if missing in the map:

```go
var indices = map[string]string{
	"UpstreamClientCredentials": "upstreamclientcredentials",
	"Upstream": "upstreams",
	"Schema": "schemas",
}
    
func toIndex(resource AgogosResource) string {
	index, ok := indices[resource.Kind]
	if !ok {
		index = unknownKind
	}
	
	return index
}
```